FabNESO/ensemble_tools.py

```diff
@@ -70,10 +78,15 @@ def _product_dict(input_dict: dict) -> Iterator[dict]:
         yield dict(zip(keys, values, strict=True))
 
 
+def indices_iterator(n_dirs: int, n_parameters: int) -> Iterator[tuple[Any, ...]]:
```

Comment:
```suggestion
def indices_iterator(n_dirs: int, n_parameters: int) -> Iterator[tuple[int, ...]]:
```

I think the tuples iterated over will always contain integers so type hint can be more specific here? Wasn't sure if `mypy` would complain about this if type hint for `itertools.product` is too broad, but from testing locally seems to work okay. Import of `Any` can also be removed in this case (along with adjustment to other `Any` usages below).

Reply:
Sure, `mypy` seems to be happy with this, so I've done as you suggest.